src/components/structures/ViewSource.tsx

```diff
@@ -77,9 +76,13 @@ export default class ViewSource extends React.Component<IProps, IState> {
                         <summary>
                             <span className="mx_ViewSource_heading">{_t("Decrypted event source")}</span>
                         </summary>
-                        <CopyableText getTextToCopy={copyDecryptedFunc}>
-                            <SyntaxHighlight language="json">{stringify(decryptedEventSource)}</SyntaxHighlight>
-                        </CopyableText>
+                        {decryptedEventSource ? (
+                            <CopyableText getTextToCopy={copyDecryptedFunc}>
+                                <SyntaxHighlight language="json">{stringify(decryptedEventSource)}</SyntaxHighlight>
+                            </CopyableText>
+                        ) : (
+                            <div>{_t("Source unavailable")}</div>
```

Comment:
Perhaps?

```suggestion
                            <div>{_t("Decrypted source unavailable")}</div>
```

Reply:
Sounds good, updated!

I double checked, and it does seem that the only reason that `clearEvent` would be `undefined` is if it was redacted, though AFAIK that's not really guaranteed anywhere.

Reply:
You're right, sorry for misleading you